File: java/rna-transcription/src/main/java/RnaTranscription.java
public class RnaTranscription {

    public String transcribe(String dnaStrand) {
        String result = "";
        for (int i = 0; i < dnaStrand.length(); i++) {
            char dnaChar = dnaStrand.charAt(i);
            switch (dnaChar) {
                case 'G':
                    result += 'C';
                    break;
                case 'C':
                    result += 'G';
                    break;
                case 'T':
                    result += 'A';
                    break;
                case 'A':
                    result += 'U';
                    break;
                case 'U':
                    result += 'A';
                    break;
            }
        }
        return result;
    }

}

File: cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"

namespace transcription {

    char to_rna(char in) {
        switch (in) {
            case 'C': return 'G';
            case 'G': return 'C';
            case 'A': return 'U';
            case 'T': return 'A';
            default: return '?';
        }
    }

    std::string to_rna(const std::string &in) {
        std::string result;
        for (const auto &i : in) {
            result += to_rna(i);
        }
        return result;
    }

}  // namespace transcription
